streams/src/test/java/org/apache/kafka/streams/integration/utils/ZooKeeperEmbedded.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.integration.utils;
+
+import org.apache.curator.test.TestingServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Runs an in-memory, "embedded" instance of a ZooKeeper server.
+ *
+ * The ZooKeeper server instance is automatically started when you create a new instance of this
+ * class.
+ */
+public class ZooKeeperEmbedded {
```

Comment:
In general, we need to check how `ZookeeperEmbedded`, `KafkaEmbedded`, and `EmbeddedSingleNodeKafkaCluster` compares to existing functionality in Kafka.  @dguy mentioned `EmbeddedZookeeper`, but I think there's more such overlapping functionality.